specification/quota/resource-manager/Microsoft.Quota/preview/2023-06-01-preview/groupquota.json

```diff
@@ -0,0 +1,1797 @@
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "GroupQuotas"
+        ],
+        "summary": "Deletes GroupQuotas for the name passed.",
+        "description": "Deletes the GroupQuotas for the name passed. All the remaining shareQuota in the GroupQuotas will be lost.",
+        "operationId": "GroupQuotas_Delete",
+        "x-ms-examples": {
+          "GroupQuotas_Delete_Request_ForCompute": {
+            "$ref": "./examples/GroupQuotas/DeleteGroupQuotas.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/MgIdInParameters"
+          },
+          {
+            "$ref": "#/parameters/GroupQuotaNameInParameters"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK."
+          },
+          "204": {
+            "description": "No Content."
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Management/managementGroups/{mgId}/providers/Microsoft.Quota/groupQuotas": {
+      "get": {
+        "tags": [
+          "GroupQuotas"
+        ],
+        "summary": "Lists GroupQuotas for the scope passed.",
+        "description": "Lists GroupQuotas for the scope passed. It will return the GroupQuotas QuotaEntity properties only.The details on groupQuota can be access from the groupQuota APIs.",
+        "operationId": "GroupQuotas_List",
+        "x-ms-examples": {
+          "GroupQuotas_List_Request_ForCompute": {
+            "$ref": "./examples/GroupQuotas/GetGroupQuotasList.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/MgIdInParameters"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/GroupQuotaList"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Management/managementGroups/{mgId}/providers/Microsoft.Quota/groupQuotas/{groupQuotaName}/subscriptions/{subscriptionId}": {
+      "put": {
```

Comment:
Here, subscriptions doesn't look a valid resource, since you are adding subscription id information to the parent resource groupQuotas itself, and not storing the subscriptions resource as a separate resource. Basically, this API should be creating/updating/reading/delete a resource corresponding to Microsoft.Quota/groupQuotas/subscriptions, whereas it is currently doing CRUD on groupQuota resource itself.

Reply:
@visingla-ms what is the recommended action here? remove this endpoint and have users just do `GroupQuota` CRUD operations on the  `"/providers/Microsoft.Management/managementGroups/{mgId}/providers/Microsoft.Quota/groupQuotas"` endpoint?

Reply:
As @visingla-ms mentioned, this path is used when creating a resource under groupQuotas, not adding/updating properties on groupQuotas. If you want to update the subscription properties on groupQuotas, you can just use the PATCH operation you had for groupQuotas

Reply:
yes, ideally we would like to be able to operate on the subscriptionList solely through the GroupQuotas PATCH API. however, we want to still have a separate REST API just to modify subscriptions membership to the GroupQuota object to make it simpler for the user to solely operate on subscriptions. we anticipate users will be more often editing this particular field of GroupQuotas.

in the instance a user were to try to update the subscriptionList via the GroupQuotas PATCH API and that GroupQuota has 1000s+ subscriptions associated with it, it becomes relatively messy for that user to modify specific subscriptions in the list. if they say wanted to remove 1 id from the list they would need to pass the other 999 ids in their request body.

retaining a separate API would would streamline the process for users of the API greatly.

this would also users to also remove multiple subscriptions in parallel/from different instance of their services

Reply:
Getting back on this after the discussion we had last week, the better way to model this is to make it a nested resource, so this path makes sense with CRUD, when you use array as properties, it also has sizing, and other race condition limitations